### PR TITLE
Update ALttP in-game hints 

### PR DESCRIPTION
**Pull request by t3hf1gm3nt**
These are proposed changes to some of the in-game hint texts for certain locations. These changes make it so each location/area have distinct hints that only refer to one location/area. (please note this is my first time EVER using git so I hope I'm doing this correctly)

### CODE REVIEW

File: worlds/alttp/Regions.py [hunk 1]
```diff
@@ -468,19 +468,19 @@ def mark_light_world_regions(world, player: int):
...
      'Mini Moldorm Cave - Right': (0xeb48, 0x18631c, False, 'near Moldorms'),
      'Mini Moldorm Cave - Far Right': (0xeb4b, 0x18631f, False, 'near Moldorms'),
      'Mini Moldorm Cave - Generous Guy': (0x180010, 0x18635a, False, 'near Moldorms'),
      'Ice Rod Cave': (0xeb4e, 0x186322, False, 'in a frozen cave'),
-     'Bonk Rock Cave': (0xeb3f, 0x186313, False, 'alone in a cave'),
+     'Bonk Rock Cave': (0xeb3f, 0x186313, False, 'in a cave with an altar'),
      'Library': (0x180012, 0x18635c, False, 'near books'),
      'Potion Shop': (0x180014, 0x18635d, False, 'near potions'),
-     'Lake Hylia Island': (0x180144, 0x186353, False, 'on an island'),
+     'Lake Hylia Island': (0x180144, 0x186353, False, 'on an island in the water'),
```
Comment (alwaysintreble) [inline worlds/alttp/Regions.py:483]: floating in the water?

**Comment by t3hf1gm3nt** [comment]
Updated dungeon boss hints to "the boss of [dungeon]" instead of the vanilla boss names

**Comment by t3hf1gm3nt** [comment]
Updated Pedestal Hint Text to remove references to vanilla bosses, opting to replace them using the naming convention the other dungeon items use.

File: worlds/alttp/Regions.py (hunk 1 above)
Comment (t3hf1gm3nt) [reply worlds/alttp/Regions.py:483]: the item isn't floating in the actual water though. could do island in the lake though possibly?

**Pull request merged**